### ISSUE TITLE
Github deskop doesn't find repo if diff of deleted files is too big

### ISSUE THREAD

**Issue by fuomag9** [desktop/desktop#9049]
### Describe the bug

Github deskop doesn't find repo if diff of deleted files is too big

### Version & OS

Github desktop 2.3.1 
Windows 10 1903 18362.535

### Steps to reproduce the behavior

1. Git clone https://github.com/yeKcim/warmux
2. Remove the `old` folder inside `warmux`
3. Open github desktop
4. Github desktop will report that it cannot find the repo, while it still exits on disk

### Expected behavior

Show the big diff since a lot of files have been removed

### Actual behavior

Github desktop will report that it cannot find the repo, while it still exits on disk

### Screenshots
Github desktop not showing the diff
![1](https://user-images.githubusercontent.com/1580624/73791406-892e2c00-47a2-11ea-92f8-ec88e0fe0219.png)

Git gui showing the diff
![2](https://user-images.githubusercontent.com/1580624/73791414-8c291c80-47a2-11ea-8150-a235aa6455fe.png)

### Logs

[2020-02-04.desktop.production.log](https://github.com/desktop/desktop/files/4156373/2020-02-04.desktop.production.log)

**Comment by tierninho** [comment]
@fuomag9 thanks for the report!

> When i look on logs i saw that :
> "error: [ui] 'git status' emitted 21525956 bytes, which is beyond the supported threshold of 20000000 bytes"

This was a change we made in #5053 to address a previous limitation with reading the status of a repository #2449 - previously it would error after encountering a `git status` output of 10MB or more.

With that change we can see how many bytes are available before we read it in. We set the limit to 20MB as that seemed like a reasonable upper bound at the time. It looks like your example repository is over 34MB of output from `git status`, which suggests thousands of untracked files and/or a missing `.gitignore` file in the repository.

As a workaround, if you make the first commit using the Git command line and then drag-and-drop the repository into the application again, that should get you back to a good place.
